src/nvim/eval.c

```diff
@@ -12245,19 +12249,18 @@ static void f_reltime(typval_T *argvars, typval_T *rettv) FUNC_ATTR_NONNULL_ALL
   // (varnumber_T is defined as int). For all our supported platforms, int's
   // are at least 32-bits wide. So we'll use two 32-bit values to store it.
   union {
-    struct { varnumber_T low, high; } split;
+    struct { uint32_t low, high; } split;
     proftime_T prof;
   } u = { .prof = res };
 
-  // statically assert that the union type conv will provide the correct
-  // results, if varnumber_T or proftime_T change, the union cast will need
-  // to be revised.
-  STATIC_ASSERT(sizeof(u.prof) == sizeof(u) && sizeof(u.split) == sizeof(u),
-      "type punning will produce incorrect results on this platform");
```

Comment:
@splinterofchaos Since you are using `proftime_T` and not `uint64_t` this assert is needed. Also note the exact expression in check: it checks *not* that `sizeof(proftime_T) == 2 * sizeof(uint32_t)`, but that they have equal size with the struct size (if I am not mistaking this is a guard against possible alignment skips: the thing I would not think about if I were writing the check, but if I thought I would use something like `sizeof(u.split) == sizeof(u.split.low)+sizeof(u.split.high) && sizeof(u.prof) == sizeof(u.split) && sizeof(u.prof) == sizeof(u)`: reasoning for additional check is that this way you can’t have `union { uint128_t prof; struct { uint32_t low; uint32_t skip1; uint32_t high; uint32_t skip2 } }`).

About varnamber_T: it is mentioned in documentation that variables have 32 or 64 bits, so I am being paranoid.